### ISSUE TITLE
Unexpected behavior of the map function, when inserting nodes into the DOM tree

### ISSUE THREAD

**Issue by napretep** [fable-compiler/Fable#3069]
### Description

Unexpected behavior of the map function, when inserting nodes into the DOM tree

In fact these nodes are not inserted into the DOM tree by the map function

### Repro code

[here is the code ](https://fable.io/repl/#?code=LYewJgrgNgpgBAQwA5IHQBcYGd0CgRIwB2cAYggEayoDCIATjPoSeVTLQxwFIDKu6AJ6EyEIgGN0ASxAkAvHF6CcMYKgCCkmUWbE4AIXogA7lhj1UAERDBcd2OjhwKEdOlkBGOArAhxEYGJ0VHFGBEwAUVhAokcAIhc3WTjcRPciD1QpIiJzABUYAA90AB4AWjizIjA4dzgkAggkFLTPVFlxKClxAGs4crgACgAzMTh4MoA+QdwnJ3FZLBBqKBAAczg4hdig51d0uCn6xubcAEoz3AcnVqIAJm84X38Y4NCYcJgo1V2E-eTUv97lkcvkiqUKlUanUKAhesQwC0gXd2hIur1+mUhqMSBNprM5tslit1pttphYnskkQjrD4dUUhc7FcYI44EYQI4fH4AkEQmFItFfmApAA3FIc4JSGoDLb0QplSV3FLXYZDM5yAlOADatw8AG5bnd9QBdLVzJwAH0mihgAEdUMBkOyQJzUMgWGAaAALKRQMC4YaDS7PXmxVAUcCCd0oBE+v01SV2AD0yZdbo9cd9-qp6S8uFT6eCmeq8ZzRqAA&html=DwCwLgtgNgfAUHUBTAhgE3gAm54ElgqbhgAOAtEgI4CuAlgG4C8ARAMID2AdmEj+QBUAnqSQtMAY268erXgA8wAenDQA3BJAoATgGcCTGmABm5ABwt4wFagwJgAIw5ohWHMAkouDFLslRfXVYGOiQAd3EwujQwEFYAFgAGRPEQJDoAc3AE5MtrT29fKyUnF3hEFUhYOCA&css=Q)

### Expected and actual results

Open devtool, you can see that the node with id=crx-root2 has no child nodes, which means that the map function is not working properly
![image](https://user-images.githubusercontent.com/7680792/184493519-d77f4659-aeed-476b-afa4-b64d2dd745eb.png)

But the result we expect should be that there are two buttons in this node
![image](https://user-images.githubusercontent.com/7680792/184493588-03c91802-9cce-4222-b83c-a5552f29e29f.png)
In other words, I think the following two pieces of code should be equivalent.
```
root.appendChild button1 
root.appendChild button2
```
and
```
[button1;button2;]
        |> Seq.map root.appendChild
```

### Related information

* Fable version: `dotnet fable --version`
* Operating system

**Comment by alfonsogarciacaro** [comment]
Thanks a lot for providing such a detailed report @napretep! I agree the result is a bit counter-intuitive but it should fine according to F#/.NET semantics. The key is that seq/IEnumerable is a **lazy** sequence so when you call this code:

```fsharp
let f ()=
    [button1;button2;]
    |> Seq.map root.appendChild
f()
``` 

Nothing happens because you're just generating a new lazy seq. The computation won't run until you do something with it, for example converting to array: `f() |> Seq.toArray`. You can also use `List.map` which is eager so the computations will happen immediately. But in any case here you are not interested in a new collection but in running some action over all the items, so `Seq.iter` looks like a better fit and will give you the expected result (we use `ignore` to discard the result of `appendChild` and appease the compiler):

```fsharp
let f ()=
    [button1;button2;]
    |> Seq.iter (root.appendChild >> ignore)
f()
```

**Comment by napretep** [comment]
thanks, I got it.
